03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
-
-#first, let's create a function
 #this function will take two rasters (r1 and r2) and subtract them
 subtRasters <- function(r1, r2){
-return(r1-r2)
+(r1-r2)
 }
-
 ```
 
+We now have a function called `subtRasters` that will subtract any two rasters
+that we give it.  
 
+##Continuing with Overlay Function
+Let's repeat our simple raster math computation using `overlay()` and the
+function we just created. 
 ```{r raster-overlay }
+CHM_ov<- overlay(DSM,DTM,fun=subtRasters)
 
-
-CHM_ov_HARV <- overlay(DSM,DTM,fun=subtRasters)
-
-plot(CHM_ov_HARV,
+plot(CHM_ov,
      main="NEON Canopy Height Model - Overlay Subtract\n Harvard Forest")
-
 ```
 
-So now, we've learned two different ways to subtract two rasters in R.
-Once we have an output raster, we can use the `writeRaster` function to export
-the object as a new geotiff. 
+#Writing Raster Files
+So now we've learned two different ways to do calculations (in our case 
+subtraction) with two rasters in R.  This gives us an output raster object 
+(`CHM_ov`), we can use the `writeRaster` function to export the object as a new 
+geotiff. 
+
+>NAMES: Part of writing a raster (or any file) is choosing a suitable name for 
+>the file.  File names should be make it clear what is in the file.  This is 
+>also true with object names in R.  In fact, so far in these lessons we've been 
+>using poor nameing practices as we haven't been including any information about
+>the location in our names (eg. `DSM` not `DSM_HARV`).  This hasn't gotten us 
+>in trouble as we've only been working in one location (Harvard Forest), 
+>however, it is still good practice to be specific with file and object names.
+
+When we write this raster object to a GeoTIFF file we'll name it 
+`chm_ov_HARV.tiff` this way we know that it is a canopy height model of Harvard 
+Forest created using overlay.  Any files you write will be saved in whichever 
+file is the current working directory. 
 
 ```{r write-raster }
-
-#export CHM object to new geotiff
-writeRaster(CHM_ov_HARV,"chm_ov_HARV.tiff",
+#export CHM object to new GeotIFF
+writeRaster(CHM_ov,"chm_ov_HARV.tiff",
             format="GTiff", 
             overwrite=TRUE, 
             NAflag=-9999)
-
-#note we are setting the NA value to -9999 for consistency.
 ```
 
-#Challenge 1
-1. Import the DSM and DTM from the SJER directory. These data are also derived from
-NEON lidar data but cover the San Joachin experimental range field site in 
-Southern California. Create a CHM from the two raster layers. Export as a geotiff.
-{: .notice }
+In the code we specify that the format will be GTiff. Overwrite is true so that
+it replaces any other file with this name.  This is a useful setting so that you
+can make a change to the object and then update the file by re-runing the code.
+However, be careful to never accidentally re-use a name for a file that already 
+exists. We are setting the NA value to -9999 for consistency. 
+####CONSISTENCY WITH WHAT?  NOT PREVIOUSLY MENTIONED
```

Comment:
Gotcha -- i thought we covered `nodata` values . We can adjust the text but in L00 we talk about no data values and for these data, they are -9999 looking at the GDAL info on them.

http://data-lessons.github.io/NEON-R-Spatial-Raster/R/Introduction-to-Raster-Data-In-R/